Automattic/jetpack
# Social: Add social post toggle component

This change adds the new component which has the checkbox for  toggling `shouldUploadAttachedMedia` that was introduced in #30179

### Other information:


## Jetpack product discussion


1204272127368434-as-1204411047112874
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Have the Advanced plan on your site, and check out this PR.
* Open a new post. In the Jetpack/Social tab there should be a new checkbox in the Media section. Toggle it on, then save post as draft. Value of the checkbox should remain after refresh.

<img width="280" alt="CleanShot 2023-04-19 at 14 51 05 png" src="https://user-images.githubusercontent.com/36671565/233080190-db01928c-d3e9-46c1-bffe-2f5cf2f00318.png">
